### PR TITLE
Link HCRP (Arc for servers) machine  to vCenter

### PR DESCRIPTION
**Pull request by nascarsayan**
---

Link HCRP machine to vCenter.
The `location`, `custom-location` properties can be figured out from the vCenter and `inventory-item` can be figured out by searching for the BiosId in the `Machine` properties (called `VMUUID`) from the `vCenter/inventoryItems`. So, making all three properties as optional.

Have added some examples in the help section for this experience.

### Related command

`az connectedvmware`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by subbartt — APPROVED** [review]

**Comment by subbartt** [comment]
@jsntcy , can you help merge the changes, seems kairu is OOF. We have validated the changes and all the tests are passing. Signed off from my side.

**Review by wangzelin007 — APPROVED** [review]

**Comment by wangzelin007** [comment]
Can you add tests for these new parameters? 
If the tests are too complex, please make sure you have tested them locally.
Others look good to me.

File: src/connectedvmware/azext_connectedvmware/_help.py [hunk 1]
```diff
@@ -392,14 +392,38 @@
...
                --name "virtual machine name" --resource-group "resource group name" \
                --vcenter "name or id of the vcenter"
+
+      - name: Enable an exiting VM to azure.
+        text: |-
+               az connectedvmware vm create --subscription contoso-sub \
+               --resource-group contoso-rg --location eastus --custom-location contoso-cl \
+               --inventory-item 01234567-0123-0123-0123-0123456789ab --name contoso-vm
+
+      - name: Link an HCRP Machine to a vCenter in another subscription.
```
Comment (wangzelin007) [inline src/connectedvmware/azext_connectedvmware/_help.py:410]:
```suggestion
      - name: Link an HCRP Machine to a vCenter in another subscription using the machine id
```

File: src/connectedvmware/azext_connectedvmware/_help.py [hunk 2]
```diff
@@ -392,14 +392,38 @@
...
+        text: |-
+               az connectedvmware vm create \
+               --machine-id /subscriptions/01234567-0123-0123-0123-0123456789ab/resourceGroups/contoso-rg/providers/Microsoft.HybridCompute/machines/contoso-vm \
+               --vcenter /subscriptions/fedcba98-7654-3210-0123-456789abcdef/resourceGroups/contoso-rg/providers/Microsoft.HybridCompute/vcenters/contoso-vcenter
+
+      - name: Link an HCRP Machine to a vCenter in another subscription.
```
Comment (wangzelin007) [inline src/connectedvmware/azext_connectedvmware/_help.py:416]:
```suggestion
      - name: Link an HCRP Machine to a vCenter in another subscription using machine name
```

**Comment by nascarsayan** [comment]
> Can you add tests for these new parameters? If the tests are too complex, please make sure you have tested them locally. Others look good to me.

@wangzelin007 , Adding automated tests would be difficult, because every time we run using `--live`, we need to revert the original VM (uninstall arc for servers), or setup a new machine. I have tested it locally, and it is working.

**Review by wangzelin007 — APPROVED** [review]

**Pull request merged**